benchmark/win.ipynb

```diff
@@ -0,0 +1,3654 @@
+   "id": "65e521f5",
+   "metadata": {
+    "papermill": {
+     "duration": 0.02811,
+     "end_time": "2023-12-11T22:19:24.985998",
+     "exception": false,
+     "start_time": "2023-12-11T22:19:24.957888",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "source": [
+    "# Constants\n",
+    "Let's configure some settings."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "id": "7414c59a",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-12-11T22:19:25.046448Z",
+     "iopub.status.busy": "2023-12-11T22:19:25.045961Z",
+     "iopub.status.idle": "2023-12-11T22:19:25.052127Z",
+     "shell.execute_reply": "2023-12-11T22:19:25.050669Z"
+    },
+    "papermill": {
+     "duration": 0.039826,
+     "end_time": "2023-12-11T22:19:25.054920",
+     "exception": false,
+     "start_time": "2023-12-11T22:19:25.015094",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "MINIMUM_MATCHES = 2\n",
+    "SEED = 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "07f60b3a",
+   "metadata": {
+    "papermill": {
+     "duration": 0.028388,
+     "end_time": "2023-12-11T22:19:25.112374",
+     "exception": false,
+     "start_time": "2023-12-11T22:19:25.083986",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "source": [
+    "## Process Data for Split\n",
+    "Let's now process the data and split it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "id": "eb55a6f0",
+   "metadata": {
+    "execution": {
+     "iopub.execute_input": "2023-12-11T22:19:25.172409Z",
+     "iopub.status.busy": "2023-12-11T22:19:25.172001Z",
+     "iopub.status.idle": "2023-12-11T22:19:26.948736Z",
+     "shell.execute_reply": "2023-12-11T22:19:26.946827Z"
+    },
+    "papermill": {
+     "duration": 1.810802,
+     "end_time": "2023-12-11T22:19:26.951729",
+     "exception": false,
+     "start_time": "2023-12-11T22:19:25.140927",
+     "status": "completed"
+    },
+    "tags": []
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Loading Raw Data into Memory\n",
+      "Processing Data:\n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "bca0363394da4b518a18fa61037eb19e",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "  0%|          | 0/61867 [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Parsed 5661 Training Matches\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(\"Loading Raw Data into Memory\")\n",
+    "\n",
+    "# Process Data\n",
+    "print(\"Processing Data:\")\n",
+    "\n",
+    "data_size = len(data)\n",
+    "\n",
+    "# Create a Progress Bar\n",
+    "t = tqdm(total=data_size)\n",
```

Comment:
Is there a reason for creating a `tqdm` instance outside loop and using `update` method rather than the more idiomatic approach of wrapping the iterable to show progress over with `tqdm`, that is:

```Python
for match in tqdm(data):
    ...
```

The current approach gives slightly confusing progress bars as the `update` method is not called if any of the conditions leading to `continue` are true, meaning the progress bar finishes with only a subset of iterations showing as complete.

Reply:
Yes, this was unintentional in this particular instance. Will be fixed in the latest change.